### DISCUSSION TITLE
Authentication rules

### DISCUSSION THREAD

**Discussion by rambant777** [wazuh/wazuh#26853] [Q&A] [answered]
Hello!
Rule 100086 shows the error “Invalid option ‘if_matched’ for rule 100086”. Can this error be due to wazuh version? (version 4.5.1). Or is the problem in the code?

Rules:
  <rule id="100085" level="3">
    <field name="win.system.eventID">4625</field>
    <description>Failed login</description>
    <group>authentication_failed</group>
  </rule>

  <rule id="100086" level="10" frequency="5" timeframe="600">
    <if_group>authentication_failed</if_group>
    <if_matched>100085</if_matched>
    <description>Multiple failed login attempts</description>
    <group>authentication_failed, multiple_attempts, brute_force</group>
  </rule>
</group>

Screen:
![Rules](https://github.com/user-attachments/assets/b2aaff67-114a-41a8-aa4f-2e7414a58a91)

**Comment by juliancnn** [comment, marked as answer]
Hi @rambant777 ,

The option you are looking for is [if_matched_sid](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#if-matched-sid), you can also check the documentation for the options [here](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html)

Regards,

**Reply by rambant777** [reply]
> Привет@rambant777,
> 
> Нужный вам параметр — [if_matched_sid](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#if-matched-sid) . Вы также можете ознакомиться с документацией по параметрам [здесь.](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html)
> 
> С уважением,

Thanks!

Now wazuh-manager does not give errors, but judging by the logs rules do not work, can you tell me what to change?

![waz](https://github.com/user-attachments/assets/f3cb9c00-51e6-4577-b176-9587ae91e715)

**Reply by juliancnn** [reply]
Hi!, 

Could you share with me the sequence of logs (from the archive.json) that you want to trigger that rule, so I can test it and look for the error?

**Reply by rambant777** [reply]
> Hi!,
> 
> Could you share with me the sequence of logs (from the archive.json) that you want to trigger that rule, so I can test it and look for the error?

Hi! Archive.json is not properly configured for logging, is the json file of rule 4625 suitable? It works, but the rules above don't log any events.
I have attached the code from the .json file of rule 4625.
[4625.json.txt](https://github.com/user-attachments/files/17829757/4625.json.txt)

**Reply by juliancnn** [reply]
Hi!,

The event you share is from windows and does not have the `source_ip field`, so rule 100085 will never match due to the `same_source_ip` condition (doc [here](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#same-srcip)). On the other hand the event is already captured by rule 4625, which is evaluated before rule 100085. If you want it can inherit from that rule with the [if_sid](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#if-sid) option.

**Reply by rambant777** [reply]
> Hi!,
> 
> The event you share is from windows and does not have the `source_ip field`, so rule 100085 will never match due to the `same_source_ip` condition (doc [here](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#same-srcip)). On the other hand the event is already captured by rule 4625, which is evaluated before rule 100085. If you want it can inherit from that rule with the [if_sid](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#if-sid) option.

Hi! 
I tried removing same_source_ip to see if the rule would work without it, but nothing changed.
In this two rule construct, I can't use if_sid instead of if_matched_sid, as I understand it?
I tried rule 4625 via wazuh-logtest and this is what I got, but there were no logs for the events (100085 and 100086):
![4625](https://github.com/user-attachments/assets/7658f3a3-6792-4ec0-b841-af5fc40c57da)

**Reply by juliancnn** [reply]
HI!, 

> In this two rule construct, I can't use if_sid instead of if_matched_sid, as I understand it?

They are different functions, in this case you only need if_matched_sid. You don't need 2 rules either, since wazuh brings the wrong loggin rule in the default ruleset, this event is captured by the rule 60122, so the only thing you need is a rule like the following one

```
   <rule id="100086" level="10" frequency="3" timeframe="600">
    <if_matched_sid>60122</if_matched_sid>
    <group>authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,</group>
    <description>Multiple loggin failed</description>
  </rule>
```

Try it in a real case, I have set the frequency to 3. You should see it in the dashborad without problems.

Let me know if it works for you.

**Reply by rambant777** [reply]
> HI!,
> 
> > In this two rule construct, I can't use if_sid instead of if_matched_sid, as I understand it?
> 
> They are different functions, in this case you only need if_matched_sid. You don't need 2 rules either, since wazuh brings the wrong loggin rule in the default ruleset, this event is captured by the rule 60122, so the only thing you need is a rule like the following one
> 
> ```
>    <rule id="100086" level="10" frequency="3" timeframe="600">
>     <if_matched_sid>60122</if_matched_sid>
>     <group>authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,</group>
>     <description>Multiple loggin failed</description>
>   </rule>
> ```
> 
> Try it in a real case, I have set the frequency to 3. You should see it in the dashborad without problems.
> 
> Let me know if it works for you.

Hi!
This rule works, thanks a lot for your help!

**Reply by rambant777** [reply]
> HI!,

HI!)
I added a line to a rule to make notifications start coming in the mail.
The notifications started coming with this sid-601222 and the spam started. Maybe you know of a feature that can be used to make it so that only notifications about custom rules 100085 (when failed login attempts are greater than 20) come in?
The pci_dss of the custom rule also matches the events that came in the mail.

**Reply by juliancnn** [reply]
Hi
> I added a line to a rule to make notifications start coming in the mail.

In the rules you have the option to form the rule to send an emai with the alert, with the option alert_by_email, you can use this option in the custom rule:

```xml
 <rule ...>
   ...
      <options>alert_by_email</options>
   ....
  </rule>
```

**Reply by rambant777** [reply]
> The pci_dss of the custom rule also matches the events that came in the mail.

Yes, I used this line for a custom rule.
The problem I described above was when I already used the alert_by_email option.
 Is it possible to do filtering on data.win.eventdata.status?

**Reply by juliancnn** [reply]
The smallest filtering option is by alerts, if you want to use the field, that field you can make a child rule. And make only one of the rules send the mails. If you wish you can tell me more about the use case so I can help you better.

**Reply by rambant777** [reply]
> The smallest filtering option is by alerts, if you want to use the field, that field you can make a child rule. And make only one of the rules send the mails. If you wish you can tell me more about the use case so I can help you better.

Hi!
I would like to be notified only when rule 100085 is triggered (When more than 20 incorrect authentication attempts), and when the rule is enabled, spam comes, not only from this rule. 
Can you please advise, maybe filtering by some of the event criteria? (as you said with a child rule).

**Reply by juliancnn** [reply]
HI rambant777 

You should analyze the logs and check what you want to ignore, for example you can make a child rule that ignores when the user is a guest:

```yml
 <rule id="100086" level="1">
    <if_sid>100085</if_sid>
    <field name="win.eventdata.targetUserName">guest</field>
    <description>Ignore failed</description>
  </rule>
```

In addition to this you can add more checks to rule 100085 which only counts failed events. For this you must determine which information or event is of interest to you and which is spam.

Regards,

**Reply by rambant777** [reply]
> In addition to this you can add more checks to rule 100085 which only counts failed events. For this you must determine which information or event is of interest to you and which is spam.

Hi!
I want to filter by data.win.eventdata.status, I selected identifiers that match the event 100085 and got the string
```
 <regex field="data.win.eventdata.status">0xc0000064|0xc000006A|0xc0000234|0xc000006d</regex> 
 ```

But still get notifications not according to the rule.
Can you please tell me how to make a child rule for notifications by this identifier?

**Reply by juliancnn** [reply]
hi @rambant777, The correct field is win.eventdata.status, the data object includes all decoded fields and is only accessible from the indexer, so your rule would be:
```xml
 <regex field="win.eventdata.status">0xc0000064|0xc000006A|0xc0000234|0xc000006d</regex> 
```

**Reply by rambant777** [reply]
>  <regex field="win.eventdata.status">0xc0000064|0xc000006A|0xc0000234|0xc000006d</regex> 
Hi @juliancnn !
Will it work if there is no such filter?
Screenshot:
![win](https://github.com/user-attachments/assets/90c2968e-fddf-4be1-b86e-2b27f4361184)

I'm thinking how to make it so that the notification only comes when there are 20+ incorrect authentication attempts under the same login. 
So far everything I've tested has failed.

**Reply by juliancnn** [reply]
HI @rambant777 ,

For this, you can use the [same_field](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#same-field) option. But considering that the original syntax problem was solved and to better organize the forum, I recommend that you create a new thread with this new problem in any of our channels (Slack, Discord, github), this way the rest of users will be able to see the threads easily and search for related problems.

Regards,

**Comment by JcabreraC** [comment]
Hi @rambant777,

As @juliancnn mentioned, the solution involves using the `same_field` option. Since the original issue has been resolved, I’ll close this thread to keep the forum organized. If you encounter further issues, feel free to start a new thread on any of our channels (Slack, Discord, GitHub) for better visibility and searchability.

Regards.
